README.md

```diff
@@ -65,55 +85,152 @@ antlr4 -Dlanguage=Python3 -visitor /tests/Expr.g4
 This will generate a number of files in the `/tests/` directory, including a Lexer (`ExprLexer.py`),
 a parser (`ExprParser.py`), and a visitor (`ExprVisitor.py`).
 
-You can use and import these directly in python. For example, from the root of this repo...
+You can use and import these directly in Python. For example, from the root of this repo:
 
 ```bash
 from tests import ExprVisitor
 ```
 
-### Creating AST Classes
+To easily use the generated files, they are put in the `antlr_py` package.
+The `__init__.py` file exports the generated files under an alias that doesn't include the name of the grammar.
+
+### Base nodes
+
+A `BaseNode` subclass has fields for all rule elements and labels for all rule element labels in its corresponding grammar rule.
+Both fields and labels are available as properties on `BaseNode` instances.
+Labels take precedence over fields if the names would collide.
+
+The name of a `BaseNode` is the name of the corresponding ANTLR grammar rule, but starting with an uppercase character.
+If rule alternative labels are specified for an ANTLR rule, these are used instead of the rule name.
+
+### Transforming nodes
+
+#### Transformer
+
+The `BaseNodeTransformer` will walk over the tree from the root node to the leaf nodes.
```

Comment:
I'm missing a potential use case of a transformer in the docs. Why would I use it?

Reply:
>>>
Typically, there is no 1-to-1 mapping between ANTLR rules and the concepts of a language and the rule hierarchy is more nested.
Transformations can be used to make the initial tree of `BaseNode`s based on ANTLR rules more similar to an AST.